Src/ILGPU.Algorithms/Runtime/Cuda/NvJpegLibrary.cs

```diff
@@ -0,0 +1,110 @@
+﻿// ---------------------------------------------------------------------------------------
+//                                   ILGPU Algorithms
+//                           Copyright (c) 2021 ILGPU Project
+//                                    www.ilgpu.net
+//
+// File: NvJpegLibrary.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details.
+// ---------------------------------------------------------------------------------------
+
+using ILGPU.Runtime.Cuda.API;
+using ILGPU.Util;
+using System;
+
+// disable: max_line_length
+
+namespace ILGPU.Runtime.Cuda
+{
+    /// <summary>
+    /// Represents an NvJpeg library.
+    /// </summary>
+    [CLSCompliant(false)]
+    public sealed partial class NvJpegLibrary : DisposeBase
+    {
+        /// <summary>
+        /// Constructs a new instance to wrap an NvJpeg library.
+        /// </summary>
+        public NvJpegLibrary(NvJpegAPI api, IntPtr libHandle)
+        {
+            API = api;
+            LibHandle = libHandle;
+        }
+
+        /// <summary>
+        /// The underlying API wrapper.
+        /// </summary>
+        public NvJpegAPI API { get; }
+
+        /// <summary>
+        /// The native handle.
+        /// </summary>
+        public IntPtr LibHandle { get; private set; }
+
+        /// <inheritdoc />
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                NvJpegException.ThrowIfFailed(
+                    API.Destroy(LibHandle));
+                LibHandle = IntPtr.Zero;
+            }
+            base.Dispose(disposing);
+        }
+
+        /// <summary>
+        /// Creates a new NvJpeg state instance.
+        /// </summary>
+        public NvJpegState CreateState()
+        {
+            NvJpegException.ThrowIfFailed(
+                API.JpegStateCreate(LibHandle, out IntPtr stateHandle));
+            return new NvJpegState(API, stateHandle);
+        }
+
+        /// <inheritdoc cref="NvJpegAPI.GetImageInfo(IntPtr, ReadOnlySpan{byte}, out int, out NvJpegChromaSubsampling, out int[], out int[])"/>
```

Comment:
Question: should we break these comments to ensure that the max-line length will not be exceeded?

Reply:
Done - didn't realise I could wrap the cref.